demos/python_demos/crnn_text_recognition_demo/crnn_text_recognition_demo.py

```diff
@@ -0,0 +1,124 @@
+                      help='Required. Path to an .xml file with a trained model.',
+                      required=True, type=str, metavar='"<path>"')
+    args.add_argument('-d', '--device',
+                      help='Optional. Specify the target device to infer on: CPU, GPU, FPGA, HDDL or MYRIAD. '
+                           'The demo will look for a suitable plugin for device specified '
+                           '(by default, it is CPU).',
+                           default='CPU', type=str, metavar='"<device>"')
+    args.add_argument('-l', '--cpu_extension',
+                      help='Required for CPU custom layers. '
+                           'Absolute path to a shared library with the kernels implementation.',
+                      default=None, type=str, metavar='"<absolute_path>"')
+    return parser
+
+def ctc_beam_search_decoder(inference_result):
+    res=np.argmax(inference_result, 2)
+    #delete repetitions
+    j=''
+    unique=[]
+    for i in np.squeeze(res):
+        if i != j:
+            unique.append(i)
+            j=i
+    #delete a delimiter (36)
+    decoded=[]
+    for i in unique:
+        if i != 36:
+            decoded.append(i)
+    return decoded
+    
+def main():
+    log.basicConfig(format='[ %(levelname)s ] %(message)s', level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+    
+    model_xml = args.model
+    model_bin = os.path.splitext(model_xml)[0] + '.bin'
+    net = IENetwork(model_xml, model_bin)
+    image_path = args.input_source
+    input_image = cv2.imread(image_path)
+    
+    log.info('Creating Inference Engine...')
+    ie = IECore() 
+    
+    if args.cpu_extension and 'CPU' in args.device:
+        ie.add_extension(args.cpu_extension, 'CPU')
+        
+    if 'CPU' in args.device:
+        supported_layers = ie.query_network(net, 'CPU')
+        not_supported_layers = [l for l in net.layers.keys() if l not in supported_layers]
+        if len(not_supported_layers) != 0:
+            log.error('Following layers are not supported by the plugin for specified device {}:\n {}'.
+                      format(args.device, ', '.join(not_supported_layers)))
+            log.error("Please try to specify cpu extensions library path in sample's command line parameters using -l "
+                      "or --cpu_extension command line argument")
+            sys.exit(1)
+
+    #load model to the plugin
+    log.info('Loading model to the plugin')
+    exec_net = ie.load_network(network=net, device_name=args.device)
+    
+    #prepare input
+    log.info("Preparing input blobs...")
+    input_blob = next(iter(net.inputs))
+    n, c, h, w = net.inputs[input_blob].shape
+    image = cv2.resize(input_image, (w, h))
+    image = image.transpose((2, 0, 1))
+    image = np.expand_dims(image,axis=0)
+
+    #run inference
+    log.info('Starting inference...')
+    res = exec_net.infer(inputs={input_blob: image})
+    out_blob = next(iter(net.outputs))
```

Comment:
You should check that the model input is of expected size: number of inputs, dimensions of a blob. The same is for `net.outputs`.